app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  # A method that updates the posts counter for a user.
```

Comment:
- [ ] OPTIONAL
It is a nice idea to use one of the Rails Active Callbacks `after_save` to call the `update_post_counter method every time a new like has been saved to database, then make the `update_post_counter` method private, similar to:

```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Hi @rloterh! Thanks for the advice I'll definitely implement this advice right away. Thanks for sharing with me about Rails Active Callbacks, they're an impressive way to handle logic before or after a change in the object state. 👍🏼